MQTTnet.Core/Client/MqttClientQueuedOptions.cs

```diff
@@ -0,0 +1,11 @@
+﻿
+
+namespace MQTTnet.Core.Client
+{
+    public class MqttClientQueuedOptions: MqttClientTcpOptions
```

Comment:
I am thinking about having two properties. So a connect method which has _MqttClientOptions_ and _MqttClientQueuedOptions_. The problem is you forgot the WebSocket options. To your client has limited functionality.

Reply:
Sorry, what you mean for "I am thinking about having two properties"?
I have to create a connect method override?

Reply:
I am not 100 % sure here. But I was thinking about a overload of _ConnectAsync_ with two types of options. So two parameters like `ConnectAsync(MqttClientQueuedOptions queuedOptions, MqttClientOptions options)`. But now I see that then we will have two _ConnectAsync_ methods because the original from the interface is also there. So maybe we have to think about the overload stuff. I would not add the _Storage_ property to the original options because it is only relevant for the queued one. If you don't have a better idea I would stick to the overload stuff like:
`MqttClientTcpOptions > MqttClientOptions` and the new one to `MqttClientQueuedTcpOptions > MqttClientTcpOptions > MqttClientOptions` and the same for WebSocket options. Then you can add the storage property (and others) and also passing the options to the internal Client (which is the field) and finally having only one _ConnectAsync_ method because the MqttQueuedClient should also implement _IMqttClient_. But I just realize that it will not work correctly. I assume we need an interface for options. Then we can extend the interface easily and having different classes with matching properties. Another idea was to move this stuff to the constructor but then the pattern is different. Options are passed in the _ConnectAsync_ method. Hard topic :smile:

And please think about renaming the client to _MqttClientManaged_ because I also have some feature requests like having the automatic connection retries and options like `UseAutoReconnect : bool` and `AutoReconnectDelay : TimeStamp`. This is not related to queued stuff so my idea is to name it `Managed`. Also the options to reflect this but not the storage methods. They are only relevant for queuing stuff.

Reply:
Modified MqttClientOptions abstract class into IMqttClientOptions, but it's not a good solution.
Must be reviewed